File: benchmark/vector_dot_product/dot.py
from array import array


def main():
    count = 6000000
    a = array("q", (0 for _ in range(count)))
    b = array("q", (0 for _ in range(count)))

    for i in range(count):
        a[i] = (i % 97) - 48
        b[i] = (i % 89) - 44

    total = 0
    for i in range(count):
        total += a[i] * b[i]

    print(f"Vector dot: {total}")


if __name__ == "__main__":
    main()

File: benchmark/vector_dot_product/dot.go
package main

import "fmt"

func main() {
	count := 6000000
	a := make([]int64, count)
	b := make([]int64, count)

	for i := 0; i < count; i++ {
		a[i] = int64((i % 97) - 48)
		b[i] = int64((i % 89) - 44)
	}

	var sum int64 = 0
	for i := 0; i < count; i++ {
		sum += a[i] * b[i]
	}

	fmt.Printf("Vector dot: %d\n", sum)
}
